avocado/utils/memory.py

```diff
@@ -250,6 +250,16 @@ def get_page_size():
     return int(output)
 
 
+def get_supported_huge_pages_size():
+    """
+    Get all supported huge page sizes for this system.
+
+    :return: list of Huge pages size (kB).
+    """
+    output = os.listdir('/sys/kernel/mm/hugepages/')
+    return [int(each.split('-')[1].strip('kB')) for each in output]
```

Comment:
`strip()` removes characters from both left and right. Because you're interested only in the right end, you can make it a bit faster by using `rstrip()`.

Reply:
About the split... you break the string into a list, then you get the list string item by index. You could make this faster by using a string slice, starting from the index right after the `-`, since the index of the `-` is fixed:
```python
>>> for file in os.listdir('/sys/kernel/mm/hugepages/'):
...     file.index('-')
...
9
```
So, just as a suggestion, already incorporating the `rstrip()`:
```python
output = os.listdir('/sys/kernel/mm/hugepages/')
# Given the items in this directory are in the format hugepages-<size>kB,
# the <size> will always start from index 10. 
return [int(each[10:].rstrip('kB')) for each in output]
```
We are talking about ~5% of performance improvement, not that much though.

Reply:
Thank you for the suggestion to improvement. I have adopted it.